### PR TITLE
Fixes #8340: logrotate promise for CFEngine Enterprise agents

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/8340

### CODE REVIEW

File: initial-promises/node-server/common/1.0/cfengine_stdlib.cf [hunk 1]
```diff
@@ -1552,6 +1552,14 @@ body file_select size_range(from,to)
 
 ##
 
+body file_select size_range(from,to)
```
Comment (peckpeck) [inline initial-promises/node-server/common/1.0/cfengine_stdlib.cf]: This looks like a duplicate of the lines above

File: initial-promises/node-server/promises.cf [hunk 2]
```diff
@@ -437,6 +437,12 @@ bundle agent garbage_collection
         file_select => days_old("30"),
         depth_search => recurse("inf");
 
+    enterprise_edition::
```
Comment (jooooooon) [inline initial-promises/node-server/promises.cf]: I don't think we should limit this to enterprise_edition. I can easily imagine Rudder being used to run on "plain" community CFEngine too. I propose we use a "fileexists("$(sys.workdir)/cf3.${sys.uqhost}.runlog")" class check to determine whether this promise should run.

File: initial-promises/node-server/common/1.0/cfengine_stdlib.cf (hunk 1 above)
Comment (ncharles) [inline initial-promises/node-server/common/1.0/cfengine_stdlib.cf]: indeed ...

File: initial-promises/node-server/promises.cf (hunk 2 above)
Comment (ncharles) [inline initial-promises/node-server/promises.cf]: ha, actually, $(sys.workdir)/promise_summary.log also need to be rotated
and on enteprise only, files:
        "$(sys.workdir)/cf_notkept.log",
        "$(sys.workdir)/cf_repair.log",
        "$(sys.workdir)/state/cf_value.log",
        "$(sys.workdir)/outputs/dc-scripts.log",
        "$(sys.workdir)/state/promise_log.jsonl",
        "$(sys.workdir)/state/classes.jsonl",

Comment (ncharles) [inline initial-promises/node-server/promises.cf]: ha, correction: only "$(sys.workdir)/cf_notkept.log", "$(sys.workdir)/cf_repair.log" exists on 3.7

**Comment by ncharles** [comment]
Commit modified

File: initial-promises/node-server/promises.cf [hunk 3]
```diff
@@ -437,6 +444,11 @@ bundle agent garbage_collection
         file_select => days_old("30"),
         depth_search => recurse("inf");
 
+      "${log_files_to_rotate}"
+        comment     => "Rotate file if above specified size",
+        rename      => rotate("10"),
+        file_select => bigger_than("1M");
```
Comment (jooooooon) [inline initial-promises/node-server/promises.cf]: Won't this fail / complain in the files are missing?

Comment (ncharles) [inline initial-promises/node-server/promises.cf]: not at all.

Comment (ncharles) [inline initial-promises/node-server/promises.cf]: /var/rudder/cfengine-community//bin/cf-agent -KIb garbage_collection 
2016-05-20T12:35:37+0200     info: Using command line specified bundlesequence
2016-05-20T12:35:37+0200     info: /default/garbage_collection/files/'/var/rudder/cfengine-community/outputs'[0]: Deleted file '/var/rudder/cfengine-community/outputs/cf_orchestrateur_1_labo_normation_com__1463135408_Fri_May_13_12_30_08_2016_0x7fcb3ef97700'
2016-05-20T12:35:47+0200     info: /default/garbage_collection/files/'/var/rudder/cfengine-community/promise_summary.log'[1]: Rotating files '/var/rudder/cfengine-community/promise_summary.log' in 10 fifo

**Pull request merged**